### DISCUSSION TITLE
TextControllerControlEditTextKt.control related crash?

### DISCUSSION THREAD

**Discussion by Matthew5716** [square/workflow-kotlin#1205] [Q&A]
I have been getting the following crash in my app for several different layouts which work as desired most of the time and it just seems crash occasionally at random. The `TextControllerControlEditTextKt.control` and `WorkflowViewStub.update` methods always seem to be contained in the stack trace. Anyone have any guidance on anything that I might be doing wrong with using the `TextControllerControlEditTextKt.control` or more info about what the root cause of the issue is. 

```
Fatal Exception: java.lang.IllegalStateException
Expected AttachedScope to be active when adding attach handler.
com.squareup.workflow1.ui.AttachedScope.runWhenAttached (ViewLaunchWhenAttached.kt:142)
com.squareup.workflow1.ui.ViewLaunchWhenAttachedKt.launchWhenAttached (ViewLaunchWhenAttached.kt:74)
com.squareup.workflow1.ui.ViewLaunchWhenAttachedKt.launchWhenAttached$default (ViewLaunchWhenAttached.kt:59)
com.squareup.workflow1.ui.TextControllerControlEditTextKt.control (TextControllerControlEditText.kt:37)
com.eliteextra.driver.common.android.stop.list.OrderLayoutRunner.showRendering (OrderLayoutRunner.kt:78)
com.eliteextra.driver.common.android.stop.list.OrderLayoutRunner.showRendering (OrderLayoutRunner.kt:27)
com.squareup.workflow1.ui.ViewBindingViewFactory$buildView$1$1.invoke (ViewBindingViewFactory.kt:29)
com.squareup.workflow1.ui.ViewBindingViewFactory$buildView$1$1.invoke (ViewBindingViewFactory.kt:25)
com.squareup.workflow1.ui.ViewShowRenderingKt.showRendering (ViewShowRendering.kt:105)
com.squareup.workflow1.ui.WorkflowViewStub.update (WorkflowViewStub.kt:201)
com.eliteextra.driver.common.android.list.WorkflowViewStubItem.bind (WorkflowViewStubItem.kt:22)
com.eliteextra.driver.common.android.list.WorkflowViewStubItem.bind (WorkflowViewStubItem.kt:10)
com.xwray.groupie.viewbinding.BindableItem.bind (BindableItem.java:72)
com.xwray.groupie.viewbinding.BindableItem.bind (BindableItem.java:50)
com.xwray.groupie.viewbinding.BindableItem.bind (BindableItem.java:23)
com.xwray.groupie.Item.bind (Item.java:49)
com.xwray.groupie.GroupAdapter.onBindViewHolder (GroupAdapter.java:233)
com.xwray.groupie.GroupAdapter.onBindViewHolder (GroupAdapter.java:20)
androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder (RecyclerView.java:7337)
```

**Comment by rjrjr** [comment]
I don't have a quick answer I'm afraid. FWIW I have never had a good experience trying to manage `EditText` instances in `RecyclerView`, with or without Workflow.

I do notice that you're using a pretty old release (`ViewBindingViewFactory` is long dead), but I don't know that anything has changed in the meantime that would improve this issue.

**Reply by Matthew5716** [reply]
Thanks for the quick response! I was afraid that would be the case.

**Reply by Matthew5716** [reply]
Out of curiosity, looks like I am on version 1.7.1 which is the last non beta release version. Do you recommend just upgrading to the latest beta version?

**Reply by rjrjr** [reply]
Yes. We use it in production. The "beta" label has lingered an embarrassingly long time as we sort out the last few nuances of our Compose interaction.
